mods/enchanteditems/rulesets.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+exports.BattleFormats = {
+	batonpassclause: {
+		effectType: 'Banlist',
+		name: 'Baton Pass Clause',
+		onStart: function () {
+			this.add('rule', 'Baton Pass Clause: Limit one Baton Passer, can\'t pass Spe and other stats simultaneously');
+		},
+		onValidateTeam: function (team, format) {
+			let BPcount = 0;
+			for (let i = 0; i < team.length; i++) {
+				if (team[i].moves.indexOf('Baton Pass') >= 0) {
+					BPcount++;
+				}
+				if (BPcount > 1) {
+					return [(team[i].name || team[i].species) + " has Baton Pass, but you are limited to one Baton Pass user by Baton Pass Clause."];
+				}
+			}
+		},
+		onValidateSet: function (set, format, setHas) {
+			if (!('batonpass' in setHas)) return;
+
+			// check if Speed is boosted
+			let speedBoosted = false;
+			for (let i = 0; i < set.moves.length; i++) {
+				let move = this.getMove(set.moves[i]);
+				if (move.boosts && move.boosts.spe > 0) {
+					speedBoosted = true;
+					break;
+				}
+			}
+			let boostSpeed = ['duskball', 'flamecharge', 'geomancy', 'motordrive', 'rattled', 'speedboost', 'steadfast', 'weakarmor', 'salacberry'];
```

Comment:
`salacberry` is Justified, so doesn't boost speed, and you haven't added the items for Motor Drive, Rattled, Steadfast or Weak Armor.